src/components/date-range-picker/date-range-picker.ts

```diff
@@ -1013,7 +1013,9 @@ export default class IgcDateRangePickerComponent extends FormAssociatedRequiredM
         slot="prefix"
         part=${calendarIcon}
         @pointerdown=${this._handleCalendarIconSlotPointerDown}
-        @click=${this.handleAnchorClick}
+        @click=${() => {
```

Comment:
```suggestion
        @click=${this.readOnly ? nothing : this.handleAnchorClick}
```
As an added bonus this will detach the existing event listener